crates/turborepo-micro-frontend/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+#[serde(rename_all = "camelCase")]
+pub struct VercelOptions {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stitch_applications_in_preview: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub bypass_deployment_protection_in_production: Option<bool>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+pub struct Application {
+    // default = true -> no routing
+    // default = false -> requires routing
+    pub default: bool,
+    pub routing: Option<ZoneRouting>,
+    pub development: Development,
+    pub production: Host,
+    pub metadata: Option<BTreeMap<String, String>>,
+    pub federation: Option<Federation>,
+    pub vercel: Option<Vercel>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Vercel {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub project_id: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub project_name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub deployment_protection_env_key: Option<String>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+pub struct Federation {
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub exposes: Vec<Module>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub uses: Vec<String>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+pub struct Module {
+    pub name: String,
+    pub path: String,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct ZoneRouting {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub asset_prefix: Option<String>,
+    pub matches: Vec<PathGroup>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct PathGroup {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub group: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub options: Option<PathConfigurationOptions>,
+    pub paths: Vec<String>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct PathConfigurationOptions {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub flag: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub route_to_default_application: Option<bool>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+pub struct Development {
+    pub local: Host,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fallback: Option<Host>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub task: Option<String>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+pub struct Host {
```

Comment:
what would you think about trying to match the names of the corresponding typescript types (which calls this `HostConfig`, not `Host`)